src/core/printing.c

```diff
@@ -79,6 +79,23 @@ int igraph_real_printf(igraph_real_t val) {
         return printf("%g", val);
     }
 }
+
+int igraph_real_printf_aligned(igraph_real_t val) {
```

Comment:
Some ideas:

 - Strings like `"Inf"` can also be printed in an aligned way using `%8s`
 - If we use `*` instead of a number in `printf`, then an additional `int` argument can specify the field width. This makes it possible to add an extra argument for this function that controls the field width instead of hard-coding 8.

Reply:
Very good point! Somehow I didn't think about using the `*`, even though I implemented it myself as practice last year 😬